builder/views.py

```diff
@@ -47,143 +47,58 @@ def user(request, username):
-def search(request, username, codelist_slug, search_slug):
-    search = get_object_or_404(
-        Search.objects.filter(slug=search_slug)
-        .prefetch_related("results")
-        .select_related("codelist"),
-        codelist__owner=username,
-        codelist__slug=codelist_slug,
-    )
-    codelist = search.codelist
-    results = search.results.all()
-    results_context = _results_context(codelist, results)
-    tables = _tabulate_subtrees(results_context)
-    url = (
-        reverse("builder:results", args=[codelist.owner.username, codelist.slug])
-        + f"?term={search.term}"
-    )
-
-    ctx = {
-        "codelist": codelist,
-        "search": search,
-        "url": url,
-        "tables": tables,
-        "ancestors_map": results_context["ancestors_map"],
-        "descendants_map": results_context["descendants_map"],
-        "editable": request.user == codelist.owner,
-    }
-    return render(request, "builder/search.html", ctx)
-
-
-@require_http_methods(["POST"])
-def results(request, username, codelist_slug):
+def codelist(request, username, codelist_slug, search_slug=None):
     codelist = get_object_or_404(DraftCodelist, owner=username, slug=codelist_slug)
+    coding_system = codelist.coding_system
 
-    if "term" in request.GET:
-        search = get_object_or_404(codelist.searches, term=request.GET["term"])
-        results = SearchResult.objects.filter(search=search)
-    else:
-        results = SearchResult.objects.filter(search__codelist=codelist)
-
-    updates = json.loads(request.body)["updates"]
-    results_context = _results_context(codelist, results)
-
-    actions.update_search(
-        results=results,
-        updates=updates,
-        code_to_status=results_context["code_to_status"],
-        ancestors_map=results_context["ancestors_map"],
-        descendants_map=results_context["descendants_map"],
-    )
-    return JsonResponse({"updates": updates})
+    code_to_status = dict(codelist.codes.values_list("code", "status"))
+    all_codes = list(code_to_status)
 
+    if search_slug is None:
+        search = None
+        codes_for_display = code_to_status
+    else:
+        search = get_object_or_404(codelist.searches, slug=search_slug)
+        codes_for_display = search.results.values_list("code__code", flat=True)
+
+    filter = request.GET.get("filter")
+    print(filter)
+    print(len(codes_for_display))
+    if filter == "included":
+        codes_for_display = {c for c in codes_for_display if "+" in code_to_status[c]}
+    elif filter == "excluded":
+        codes_for_display = {c for c in codes_for_display if "-" in code_to_status[c]}
+    elif filter == "unresolved":
+        codes_for_display = {c for c in codes_for_display if code_to_status[c] == "?"}
+    elif filter == "in-conflict":
+        codes_for_display = {c for c in codes_for_display if code_to_status[c] == "!"}
+        filter = "in conflict"
+    else:
+        codes_for_display = set(codes_for_display)
```

Comment:
It looks like the only thing you're changing is the conditional in the comprehension, so could probably use the dispatch pattern?

Reply:
This whole function is going to be rewritten, so I won't do that now.